src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -667,12 +690,17 @@ private void ThrowIfNoFunctionResultsAdded(IList<ChatMessage>? messages)
     /// <param name="iteration">The iteration number of how many roundtrips have been made to the inner client.</param>
     /// <param name="functionCallIndex">The 0-based index of the function being called out of <paramref name="callContents"/>.</param>
     /// <param name="captureExceptions">If true, handles function-invocation exceptions by returning a value with <see cref="FunctionInvocationStatus.Exception"/>. Otherwise, rethrows.</param>
+    /// <param name="isStreaming">Whether the function calls are being processed in a streaming context.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests.</param>
     /// <returns>A value indicating how the caller should proceed.</returns>
-    private async Task<FunctionInvocationResult> ProcessFunctionCallAsync(
-        List<ChatMessage> messages, ChatOptions options, List<FunctionCallContent> callContents,
-        int iteration, int functionCallIndex, bool captureExceptions, CancellationToken cancellationToken)
+    protected virtual async Task<FunctionInvocationResult> ProcessFunctionCallAsync(
```

Comment:
Does this definitely have to be made `public virtual`?

Looking at your implementation of `KernelFunctionInvokingChatClient`, it seems like you override `ProcessFunctionCallsAsync` and that calls `ProcessFunctionCallAsync`, which never calls `base.ProcessFunctionCallsAsync`. I might be missing something but it seems like this pattern doesn't require `ProcessFunctionCallAsync` to become protected or virtual.

Reply:
That's a good point, I might be able to reuse the function calling entirely if a CreateContext override allow me to do the 

```
var context = new AutoFunctionInvocationContext(options)
        {
            AIFunction = function,
            Arguments = new KernelArguments(callContent.Arguments ?? new Dictionary<string, object?>()) { Services = this.FunctionInvocationServices },
            Messages = messages,
            CallContent = callContent,
            Iteration = iteration,
            FunctionCallIndex = functionCallIndex,
            FunctionCount = callContents.Count,
            IsStreaming = isStreaming
        };
```

All else is similar.